### DISCUSSION TITLE
Does pnpx run binaries of installed packages?

### DISCUSSION THREAD

**Discussion by mikib0** [pnpm/pnpm#5940] [Q&A] [answered]
I want to know if pnpx is actually equivalent to npx. Because npx can run binaries from installed packages while according to what i [experienced](https://github.com/sverweij/dependency-cruiser/issues/725#issue-1533896209), pnpx doesn't. So I want to confirm if my suspicion is true.

**Comment by zkochan** [comment, marked as answer]
To run binaries, run `pnpm exec` or just `pnpm`. So to run eslint you can do `pnpm eslint` or `pnpm exec eslint`.

Use `pnpx` (or `pnpm dlx`) to download and execute a package.

So pnpx is not an equivalent of npx. pnpm has to commands for what npx does. For some cases you use `pnpm dlx` for others `pnpm exec`
